tests/benchmarks/test_micro_benchmarks.py

```diff
@@ -669,6 +670,60 @@ def validate_with_expected_error():
         benchmark(validate_with_expected_error)
 
 
+class TestBenchmarkUUID:
+    @pytest.fixture(scope='class')
+    def core_validator(self):
+        class CoreModel:
+            __slots__ = '__dict__', '__pydantic_fields_set__', '__pydantic_extra__', '__pydantic_private__'
+
+        return SchemaValidator(
+            {
+                'type': 'model',
+                'cls': CoreModel,
+                'schema': {
+                    'type': 'model-fields',
+                    'fields': {'u': {'type': 'model-field', 'schema': {'type': 'uuid'}}},
+                },
+            }
+        )
+
+    @pytest.fixture(scope='class')
+    def uuid_raw(self):
+        return UUID('12345678-1234-5678-1234-567812345678')
+
+    @pytest.fixture(scope='class')
+    def uuid_str(self, uuid_raw):
+        return str(uuid_raw)
+
+    @pytest.fixture(scope='class')
+    def python_data_dict(self, uuid_raw):
+        return {'u': uuid_raw}
+
+    @pytest.fixture(scope='class')
+    def json_dict_data(self, uuid_str):
+        return json.dumps({'u': uuid_str})
+
+    @pytest.mark.benchmark(group='uuid model - python')
+    def test_core_python(self, core_validator, benchmark, python_data_dict):
+        benchmark(core_validator.validate_python, python_data_dict)
+
+    @pytest.mark.benchmark(group='uuid model - JSON')
+    def test_model_core_json(self, core_validator, benchmark, json_dict_data):
+        benchmark(core_validator.validate_json, json_dict_data)
+
+    @pytest.mark.benchmark(group='uuid uuid')
+    def test_core_raw(self, benchmark, uuid_raw):
+        v = SchemaValidator({'type': 'uuid'})
+
+        benchmark(v.validate_python, uuid_raw)
+
+    @pytest.mark.benchmark(group='uuid str')
+    def test_core_str(self, benchmark, uuid_str):
+        v = SchemaValidator({'type': 'uuid'})
+
+        benchmark(v.validate_python, uuid_str)
```

Comment:
It would be great to compare these benchmarks against the current Pydantic implementation (in Python), if you're willing to do that? That would confirm our assumption that migrating this to Rust is worth it :)

It should be possible to do that by building the same core schema as Pydantic currently does (maybe with some copy-paste) and feeding that to a validator.

Reply:
Good idea, do you have an example to create this benchmark?

Reply:
Let me update #763 later with a similar benchmark and get back to you :)

Reply:
Done d9e45c14509f78dbdb737e4a77c313b26c80980c, WDYT?

Reply:
Looks great! So I see we're about 33% faster to have a Rust implementation, makes sense to me as the heavy lifting in the Python UUID implementation will be implemented in C anyway; what we're shaving off is the dynamic overheads of checking & constructing a UUID.